runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java

```diff
@@ -0,0 +1,438 @@
+			menuEntry.setTarget(event.getTarget());
+			menuEntry.setType(MenuAction.CANCEL.getId());
+
+			client.setMenuEntries(menuEntries);
+		}
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (!event.getMenuOption().equals(MARK))
+		{
+			return;
+		}
+
+		MenuEntry[] menuEntries = client.getMenuEntries();
+
+		// get tile coordinates off of walk here entry
+		for (int i = 0; i < menuEntries.length; i++)
+		{
+			MenuEntry menuEntry = menuEntries[i];
+			if (menuEntry.getOption().equals(WALK_HERE))
+			{
+				Point target = new Point(menuEntry.getParam0(), menuEntry.getParam1());
+				markTile(target);
+				break;
+			}
+		}
+	}
+
+	@Override
+	protected void startUp()
+	{
+		keyManager.registerKeyListener(inputListener);
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		keyManager.unregisterKeyListener(inputListener);
+	}
+
+	@Override
+	public GroundMarkerOverlay getOverlay()
+	{
+		return overlay;
+	}
+
+	protected void markTile(Point ptMouse)
+	{
+		LocalPoint localPoint = getTile(ptMouse);
+
+		if (localPoint == null)
+		{
+			return;
+		}
+
+		WorldPoint worldPoint;
+
+		if (client.isInInstancedRegion())
+		{
+			// get position in the scene
+			int sceneX = localPoint.getRegionX();
+			int sceneY = localPoint.getRegionY();
+
+			// get chunk from scene
+			int chunkX = sceneX / CHUNK_SIZE;
+			int chunkY = sceneY / CHUNK_SIZE;
+
+			// get the template chunk for the chunk
+			int[][][] instanceTemplateChunks = client.getInstanceTemplateChunks();
+			int templateChunk = instanceTemplateChunks[client.getPlane()][chunkX][chunkY];
+
+			int rotation = templateChunk >> 1 & 0x3;
+			int templateChunkY = (templateChunk >> 3 & 0x7FF) * CHUNK_SIZE;
+			int templateChunkX = (templateChunk >> 14 & 0x3FF) * CHUNK_SIZE;
+			int plane = templateChunk >> 24 & 0x3;
+
+			// calculate world point of the template
+			int x = templateChunkX + (sceneX & (CHUNK_SIZE - 1));
+			int y = templateChunkY + (sceneY & (CHUNK_SIZE - 1));
+
+			worldPoint = new WorldPoint(x, y, plane);
+			// rotate point back to 0, to match with template
+			worldPoint = rotate(worldPoint, rotation);
+		}
+		else
+		{
+			worldPoint = WorldPoint.fromLocal(client, localPoint);
+		}
+
+		int regionX = worldPoint.getX() >> 6;
+		int regionY = worldPoint.getY() >> 6;
+		int regionId = regionX << 8 | regionY;
+		GroundMarkerPoint point = new GroundMarkerPoint(regionId, worldPoint.getX() & 0x3f, worldPoint.getY() & 0x3f, client.getPlane());
+		log.debug("Updating point: {} - {}", point, worldPoint);
+
+		List<GroundMarkerPoint> points = new ArrayList<>(getPoints(regionId));
+		if (points.contains(point))
+		{
+			points.remove(point);
+		}
+		else
+		{
+			points.add(point);
+		}
+
+		savePoints(regionId, points);
+
+		loadPoints();
+	}
+
+	/**
+	 * Rotate the chunk containing the given point to rotation 0
+	 *
+	 * @param point
+	 * @param rotation
+	 * @return
+	 */
+	private WorldPoint rotate(WorldPoint point, int rotation)
```

Comment:
What is the difference between rotate and rotateChunk? They look like duplicates